### PR TITLE
NC-625

### PR DESCRIPTION
**Pull request by Joeper214**
ref: https://skyarch.backlog.jp/view/NC-625

### CODE REVIEW

**Comment by Joeper214** [comment]
Please review and Merge PR. @root-beer

**Comment by Joeper214** [comment]
Please review and merge PR. @root-beer  Thanks!

File: app/controllers/key_pairs_controller.rb [hunk 1]
```diff
@@ -39,9 +40,15 @@ def destroy
       secret_access_key: @project.secret_access_key,
       region:            region
     )
-    ec2.delete_key_pair(key_name: key_name)
+    keys = ec2.describe_key_pairs()
+    keys[:key_pairs].each do |item|
+      if item.key_fingerprint.eql? key_name
```
Comment (root-beer) [inline app/controllers/key_pairs_controller.rb]: I think the comparison of `key_fingerprint` and `key_name` is strange.

Comment (Joeper214) [inline app/controllers/key_pairs_controller.rb]: Yeah I think so too. Sorry I'll  rewrite this functionality and Mark it as WIP.

**Comment by Joeper214** [comment]
I rewritten the methods. I think you can review this now. @root-beer  Thanks.

File: app/controllers/key_pairs_controller.rb [hunk 2]
```diff
@@ -51,4 +54,15 @@ def destroy
   def set_project
     @project = Project.find(params.require(:project_id))
   end
+
+  def check_fingerprint(fingerprint)
+    keys = @ec2.describe_key_pairs()
+    keys[:key_pairs].each do |item|
+      if item.key_fingerprint.eql? fingerprint
+        @key_name = item.key_name
```
Comment (root-beer) [inline app/controllers/key_pairs_controller.rb]: It should break after the substitution.

Comment (Joeper214) [inline app/controllers/key_pairs_controller.rb]: :+1:

**Comment by root-beer** [comment]
LGTM. Thanks!

**Pull request merged**